discovery/pkg/sync/endpoints.go

```diff
@@ -28,90 +28,83 @@ import (
+func AddEndpointsAction(endpoints *v1.Endpoints, upstreamName string) Action {
+	return endpointsAction{kind: actionAdd, upstreamName: upstreamName, endpoints: endpoints}
 }
 
 // UpdateEndpointsAction returns an action that updates the given endpoint in the cluster
-func UpdateEndpointsAction(endpoints *v1.Endpoints) Action {
-	return endpointsAction{kind: actionUpdate, endpoints: endpoints}
+func UpdateEndpointsAction(endpoints *v1.Endpoints, upstreamName string) Action {
+	return endpointsAction{kind: actionUpdate, upstreamName: upstreamName, endpoints: endpoints}
 }
 
 // DeleteEndpointsAction returns an action that deletes the given endpoint from the cluster
-func DeleteEndpointsAction(endpoints *v1.Endpoints) Action {
-	return endpointsAction{kind: actionDelete, endpoints: endpoints}
+func DeleteEndpointsAction(endpoints *v1.Endpoints, upstreamName string) Action {
+	return endpointsAction{kind: actionDelete, upstreamName: upstreamName, endpoints: endpoints}
 }
 
 // endpointsAction is an action that is to be performed on a specific endpoint.
 type endpointsAction struct {
-	kind      string
-	endpoints *v1.Endpoints
+	kind         string
+	endpoints    *v1.Endpoints
+	upstreamName string
 }
 
 // ObjectMeta returns the objectMeta piece of the Action interface object
 func (action endpointsAction) ObjectMeta() *metav1.ObjectMeta {
 	return &action.endpoints.ObjectMeta
 }
 
-// Sync performs the action on the given Endpoints resource
-func (action endpointsAction) Sync(kubeClient kubernetes.Interface, metrics localmetrics.DiscovererMetrics, backendName string) error {
+func (action endpointsAction) GetActionType() string {
+	return action.kind
+}
 
+// Sync performs the action on the given Endpoints resource
+func (action endpointsAction) Sync(kubeClient kubernetes.Interface, logger *logrus.Logger) error {
 	var err error
 	switch action.kind {
 	case actionAdd:
-		err = addEndpoints(kubeClient, action.endpoints, metrics, backendName)
+		err = addEndpoints(kubeClient, action.endpoints)
 	case actionUpdate:
-		err = updateEndpoints(kubeClient, action.endpoints, metrics, backendName)
+		err = updateEndpoints(kubeClient, action.endpoints)
 	case actionDelete:
-		err = deleteEndpoints(kubeClient, action.endpoints, metrics, backendName)
+		err = deleteEndpoints(kubeClient, action.endpoints)
 	}
 	if err != nil {
 		return fmt.Errorf("error handling %s: %v", action, err)
 	}
 
-	metrics.EndpointsEventTimestampMetric(action.endpoints.GetNamespace(), backendName, action.endpoints.GetName(), time.Now().Unix())
 	return nil
 }
 
 func (action endpointsAction) String() string {
 	return fmt.Sprintf(`%s endpoints '%s/%s'`, action.kind, action.endpoints.Namespace, action.endpoints.Name)
 }
 
-func addEndpoints(kubeClient kubernetes.Interface, endpoints *v1.Endpoints, lm localmetrics.DiscovererMetrics, backendName string) error {
+func (action endpointsAction) LogMetrics(gimbalKubeClient kubernetes.Interface, metrics localmetrics.DiscovererMetrics,
```

Comment:
nit: Can we call this `SetMetrics` or `UpdateMetrics`? Log makes me think of log messages

Reply:
How about `WriteMetrics`? Set & Update sound like they are values that are getting set.

Reply:
eh we can do `SetMetrics` I think